src/test/java/org/folio/rest/api/AuthorityStorageTest.java

```diff
@@ -1,39 +1,42 @@
 package org.folio.rest.api;
 
 import static java.util.concurrent.TimeUnit.SECONDS;
-import static org.junit.Assert.assertEquals;
-
-import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
 import static org.folio.rest.support.http.InterfaceUrls.authoritiesStorageUrl;
 import static org.folio.rest.support.matchers.DomainEventAssertions.assertCreateEventForAuthority;
 import static org.folio.rest.support.matchers.DomainEventAssertions.assertRemoveAllEventForAuthority;
 import static org.folio.rest.support.matchers.DomainEventAssertions.assertRemoveEventForAuthority;
 import static org.folio.rest.support.matchers.DomainEventAssertions.assertUpdateEventForAuthority;
+import static org.folio.utility.RestUtility.TENANT_ID;
+import static org.folio.utility.VertxUtility.getClient;
+import static org.junit.Assert.assertEquals;
 
+import io.vertx.core.json.JsonObject;
 import java.net.HttpURLConnection;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
-
-import io.vertx.core.json.JsonObject;
 import junitparams.JUnitParamsRunner;
 import lombok.SneakyThrows;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-
 import org.folio.rest.jaxrs.model.Authority;
 import org.folio.rest.support.Response;
 import org.folio.rest.support.ResponseHandler;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
 
 @RunWith(JUnitParamsRunner.class)
 public class AuthorityStorageTest extends TestBase {
 
+  @SneakyThrows
   @Before
   public void beforeEach() {
+    clearData();
     StorageTestSuite.deleteAll(authoritiesStorageUrl(""));
```

Comment:
Why are authorities deleted separately, are they not included in the `clear data` operation?

Reply:
> Why are authorities deleted separately, are they not included in the clear data operation?

This is one of the problems I lost a lot of time fighting with.
The `clearData()` does not clear everything.
I've attempted to delete all data before all tests to maintain clean state.
Tests across several test classes do not like this.
They are too interdependent on other tests and data or the delete process needs to be far more complex than it currently is.

I essentially gave up and resigned to just have the clearAll() delete all data in such a way that tests don't suddenly fail because the data was deleted.

There is a huge problem with isolation of the data and I essentially found it too unweildly to try and solve.
I only eased the isolation pains with some of the changes rather than solving them.

Reply:
I've added additional documentation as requested.